### PR TITLE
Feature/211

### LINKED ISSUE #211: Implement response caching decorator @cache

### PR DESCRIPTION
**Pull request by Ghnuberath**
Implementation of `@cache` decorator for transparent response caching. Fixes #211

### CODE REVIEW

File: lib/util/response_cache.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  /**
+   * Wraps get in a Promise.
+   *
+   * @param {string} key - The key to get.
+   * @private
+   */
+  get (key) {
+    return new Promise((resolve, reject) => {
+      this[sRavelInstance].kvstore.get(key, (err, result) => {
+        this[sRavelInstance].log.trace(`Could not get cached response for key ${key}`);
```
Comment (plaliberte) [inline lib/util/response_cache.js]: I think this will fire on a cache hit as well

Comment (Ghnuberath) [reply lib/util/response_cache.js]: Whoops! yup.

File: lib/util/response_cache.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+  /**
+   * Computes a unique hash for a string.
+   *
+   * @param {string} str - A koa context.
+   * @returns {string} A hash for the request url string.
+   * @private
+   */
+  md5 (str) {
+    return crypto.createHash('md5').update(str).digest('hex');
```
Comment (mlaccetti) [inline lib/util/response_cache.js]: I know we're not signing anything, but should we not worry about collisions?

**Comment by Ghnuberath** [comment]
It's a fine line to walk. I think knowing when to invalidate the cache requires a level of application awareness that something simple like this can't have. I'll give it some further thought, but let's move ahead with this simple caching approach for now.

**Pull request merged**